src/Microsoft.VisualStudio.Web.CodeGeneration.Msbuild/ProjectContextWriter.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.ProjectModel;
+using Microsoft.Extensions.ProjectModel.Resolution;
+using Microsoft.VisualStudio.Web.CodeGeneration.Utils;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.Msbuild
+{
+    public class ProjectContextWriter : Build.Utilities.Task
```

Comment:
I totally get why we have to do this -- dropping project.json has made this necessary-- but it seems like the complexity of this is grown so much that I think the question "Does this feature make sense as a CLI tool?" is worth considering. If scaffolding were a VS extension instead, it could lean on VS to provide this information instead of needing to add a custom task inside the user's project.